zkwrapper/zkwrapper.cc

```diff
@@ -0,0 +1,534 @@
+//
+// Created by Prudhvi Boyapalli on 10/3/16.
+//
+//
+// Modified by Zhouhan Chen on 10/4/16.
+//
+
+#include "zkwrapper.h"
+
+const std::string ZKWrapper::CLASS_NAME = ": **ZkWrapper** : ";
+
+int init = 0;
+zhandle_t *zh;
+clientid_t myid;
+
+const std::vector <std::uint8_t>
+ZKWrapper::EMPTY_VECTOR = std::vector<std::uint8_t>(0);
+
+const std::map<int, std::string> ZKWrapper::error_message = {
+	{0, "ZOK"},
+	{-1, "ZSYSTEMERROR"},
+	{-2, "ZRUNTIMEINCONSISTENCY"},
+	{-3, "ZDATAINCONSISTENCY"},
+	{-4, "ZCONNECTIONLOSS"},
+	{-5, "ZMARSHALLINGERROR"},
+	{-6, "ZUNIMPLEMENTED"},
+	{-7, "ZOPERATIONTIMEOUT"},
+	{-8, "ZBADARGUMENTS"},
+	{-9, "ZINVALIDSTATE"},
+	{-100, "ZAPIERROR"},
+	{-101, "ZNONODE"},
+	{-102, "ZNOAUTH"},
+	{-103, "ZBADVERSION"},
+	{-108, "ZNOCHILDRENFOREPHEMERALS"},
+	{-110, "ZNODEEXISTS"},
+	{-111, "ZNOTEMPTY"},
+	{-112, "ZSESSIONEXPIRED"},
+	{-113, "ZINVALIDCALLBACK"},
+	{-114, "ZINVALIDACL"},
+	{-115, "ZAUTHFAILED"},
+	{-116, "ZCLOSING"},
+	{-117, "ZNOTHING"},
+	{-118, "ZSESSIONMOVED"},
+	{-120, "ZNEWCONFIGNOQUORUM"},
+	{-121, "ZRECONFIGINPROGRESS"},
+	{-999, "ZKWRAPPERDEFAULTERROR"},
+};
+
+/**
+ * TODO
+ *
+ * @param zzh zookeeper handle
+ * @param type type of event
+ * @param state state of the event
+ * @param path path to the watcher node
+ * @param watcherCtx the state of the watcher
+ */
+void watcher(zhandle_t *zzh,
+		int type,
+		int state,
+		const char *path,
+		void *watcherCtx) {
+	LOG(INFO) << "[Global watcher] Watcher triggered on path '" << path << "'"
+		;
+	char health[] = "/health/datanode_";
+	if (type == ZOO_SESSION_EVENT) {
+		if (state == ZOO_CONNECTED_STATE) {
+			return;
+		} else if (state == ZOO_AUTH_FAILED_STATE) {
+			zookeeper_close(zzh);
+			exit(1);
+		} else if (state == ZOO_EXPIRED_SESSION_STATE) {
+			zookeeper_close(zzh);
+			exit(1);
+		}
+	}
+}
+
+
+watcher_fn ZKWrapper::watcher_health_factory(std::string path){
```

Comment:
Yes, path is being used, zk_nn_client will call zk->watcher_health_factory(ZkClientCommon::HEALTH_BACKSLASH). I can't use ZkClientCommon::HEALTH_BACKSLASH directly in zkwrapper because the variable is out of the scope.